### ISSUE TITLE
Collection quests objective order

### ISSUE THREAD

**Issue by exectails** [aura-project/aura#230]
Our quests are currently designed for sequential completion of objectives, so the next ones are revealed as you finish the previous ones.

- [X] Do something
- [ ] Do something else
- [ ] ???

Collection quests should work differently, in that you can finish the objectives in any order, with all of them being visible from the start. For example, the Black Orb collection quest, which you can finish in any order, with all fragment objectives being visible.

The quest *will* be completable once you have all items on Aura, but the objectives won't be visible or marked as done unless you have the first item, the second, the third, and so on. We could hardcode this behavior for Collect type quests, but I'd rather at a new property for it, defining how the quest should behave, in case we have to make a quest a Collect type quest (the type is sent to the client), even though it's not supposed to behave like this.

To accomplish this, the objective check logic has to be revised, to either check the current objective, or all of them, depending on the set behavior.

**Comment by exectails** [comment]
Non-collection example: Party quest "Hunt Down the Dire Sprites", 3 kill objectives for all 3 sprite types, that don't have to be done in order.

**Comment by exectails** [comment]
Since this will require some changes, we can look into properly fixing #390 on the way. Once an objective has been completed, we have to check if the next one might be done already, e.g collecting something, reaching a certain skill rank, etc.

**Comment by valdeza** [comment]
Also, said "Collect type" quests should possibly remove items specified for each Collect quest objectives and not just the last ones ([a non-Collect objective could interrupt the item removal process](https://github.com/aura-project/aura/blob/08e42a5891cc8204802b26359344f64b88227f26/src/ChannelServer/World/Entities/Creatures/CreatureQuests.cs#L415)). 
Perhaps we should add an optional boolean parametre that flags an item to stay in inventory after completing a quest. Can't think of an example off the top of my head, but I get the feeling that would be needed in the future.

**Comment by exectails** [comment]
That's not really part of this issue, but you're right of course. I did mention this somewhere, although I can't find it right now. Same for delivering. For this to work we need a callback for finishing objectives, which we currently don't have.

Ah, I remembered. I knew I mentioned it *somewhere*^^
https://github.com/aura-project/aura/blob/master/src/ChannelServer/Scripting/Scripts/QuestScript.cs#L1036
